From 69399c29963f2ceddd3210fcb87f91d9272edd08 Mon Sep 17 00:00:00 2001
From: Paul Eggleton <paul.eggleton@linux.intel.com>
Date: Sun, 15 Apr 2018 18:49:09 +0200
Subject: [PATCH] Add excludefs config option to mount module

Adds a configuration option (currently hidden) to allow the distro to
hide certain filesystems from the mount module within Webmin (e.g. /dev)
since these shouldn't be modified from the web interface.

Signed-off-by: Paul Eggleton <paul.eggleton@linux.intel.com>

Upstream-status: Pending

---
 mount/index.cgi | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/mount/index.cgi b/mount/index.cgi
index e53b468..b45221a 100755
--- a/mount/index.cgi
+++ b/mount/index.cgi
@@ -33,10 +33,12 @@ elsif ($config{'sort_mode'} == 1) {
 	}
 
 # Build visible filesystems list
+@excludefs = split(/,/, $config{excludefs});
 foreach $m (@all) {
 	@minfo = @$m;
 	$p = &simplify_mount_path($minfo[0], $minfo[2]);
 	next if ($ignore{$minfo[2]});
+	next if (grep {$_ eq $minfo[2]} @excludefs);
 	@mmodes = &mount_modes($minfo[2], $minfo[0], $minfo[1]);
 	$canedit = $can_edit{$minfo[2]} && !$mmodes[4] &&
             	   &can_edit_fs(@minfo);
